examples/hal/compute/main.rs

```diff
@@ -183,12 +183,16 @@ fn create_buffer<B: Backend>(
     let buffer = gpu.device.create_buffer(stride * len, usage).unwrap();
     let requirements = gpu.device.get_buffer_requirements(&buffer);
 
-    let ty = memory_types.into_iter().find(|memory_type| {
-        requirements.type_mask & (1 << memory_type.id) != 0 &&
-        memory_type.properties.contains(properties)
-    }).unwrap();
-
-    let memory = gpu.device.allocate_memory(&ty, requirements.size).unwrap();
+    let ty = memory_types
```

Comment:
we could at least wrap that index in a newtype thing for cleaner code

Reply:
I tend to agree but at which API level (examples, render, hal) should this be added?

Reply:
HAL is fine here. It doesn't add a ton of mental overhead, protects from ambiguation, and is still easy to escape from if the newtype contents are public.

Reply:
So, something along
```rust
// hal
pub struct MemTypeId(pub usize);
impl Device {
    fn allocate_memory(&self, id: MemTypeId, size: u64);
}

// example
let ty = MemTypeId(
     mem_types.iter().position(|...|).unwrap()
);
```
?

Reply:
yes, this will do.
The ergonomics of finding such a type ID is a little wordy, so we'll have a simple helper anyway